app/controllers/applicants_controller.rb

```diff
@@ -48,4 +21,12 @@ def creation_service
   def input
     @input ||= JSON.parse(request.raw_post, symbolize_names: true)
   end
+
+  def json_validator
+    @json_validator ||= JsonValidator.new(schema, request.raw_post)
+  end
+
+  def schema
+    @schema ||= "applicant"
+  end
```

Comment:
I think `schema_name` should be used rather than `schema` to make it clear that it's just the name your're passing in and not the schema itself.

Also, I don't think there is any benefit in  momoizing the name of the schema.  The controller class will be instantiated for each incoming request, and just process one request, so it's a bit of an overkill. So I think lines 25 to 31 can simply be replaced with: 

```
def json_validator
  JsonValidator.new('applicant', request.raw_post)
end
```